From patchwork Fri Aug  9 07:24:14 2019
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Naresh Kamboju <naresh.kamboju@linaro.org>
X-Patchwork-Id: 11085519
Return-Path: <kvm-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 56FAA912
	for <patchwork-kvm@patchwork.kernel.org>;
 Fri,  9 Aug 2019 07:24:29 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 3E12E28C2B
	for <patchwork-kvm@patchwork.kernel.org>;
 Fri,  9 Aug 2019 07:24:29 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id 3251728C3B; Fri,  9 Aug 2019 07:24:29 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-8.0 required=2.0 tests=BAYES_00,DKIM_SIGNED,
	DKIM_VALID,DKIM_VALID_AU,MAILING_LIST_MULTI,RCVD_IN_DNSWL_HI
	autolearn=unavailable version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 6EECE28C2B
	for <patchwork-kvm@patchwork.kernel.org>;
 Fri,  9 Aug 2019 07:24:25 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S2405686AbfHIHYX (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Fri, 9 Aug 2019 03:24:23 -0400
Received: from mail-wm1-f68.google.com ([209.85.128.68]:54918 "EHLO
        mail-wm1-f68.google.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S1727040AbfHIHYW (ORCPT <rfc822;kvm@vger.kernel.org>);
        Fri, 9 Aug 2019 03:24:22 -0400
Received: by mail-wm1-f68.google.com with SMTP id p74so4646246wme.4
        for <kvm@vger.kernel.org>; Fri, 09 Aug 2019 00:24:21 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=linaro.org; s=google;
        h=from:to:cc:subject:date:message-id;
        bh=C5YgxiRdWg1QeF0RKI1Qt9EbdjzocEjmRbp/WRepaaE=;
        b=y6iGPlilN8Si8QhpWCJ9jAq3We6ZMX7x6l04tVExxbpkgl4qKg9Dj98hDdVl7XWf8y
         PV3WfvEwgUvjE7oGBt/zg5N+39uCB7dUHoUdKPCtiLHMgrKvCYxNX2PhKBSjqFWwgGzp
         6Qz/slTlHhh8Hrpf6CVmmYFnCBm0Xd/j2crNXq78Dy4j9qQDyafXTpX1rD1XFAwPTqyA
         eKbxucva1RMOlQQ6OpUGTEpxL29e5/jioOouxU4KYUjFKudIJaxLhzNkDHqieMfkZNOU
         E3HlzzQ9vbOJSbVzD6y9s/gyAQpxQDe+yyf4X2R6KRsl5tWkHBNOY00kybUw7UWKOdCx
         mHUQ==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:from:to:cc:subject:date:message-id;
        bh=C5YgxiRdWg1QeF0RKI1Qt9EbdjzocEjmRbp/WRepaaE=;
        b=rXz7/yhsHJ7O7AOAcxaz9JjhZEdFogcfnQAIHTJAsNWK+RoKIZLGAI3cub/UAg3gVR
         figNL+bgapB+DydKMouD1OPy6CVZfoC2CZF77na/T9OODrrPKF/zsnnb2N+CiHtqfyHD
         d777AylwDJJwJFLhoX+IGH8HLhENI3ni1E5TTDOO5FOFBjtVUkPljEijKZ28pNPuZy6s
         nx1j5MyqwshrkRE/Pw6MJbdoCvSg2VaR1P3tLSFRstxDCOvVUHYNYI0YQna5HKsXgwgo
         kQ3J0OxSgIjtAuI9YIDU65ObzLgGi/OAtHUSlk0PpRz7k/82+bsqLmvzqsAFtYKkRwPz
         WL3Q==
X-Gm-Message-State: APjAAAVW7+IWEwi2aMpRghKdtzSdhbeT7cBJieoGzbAZIlWO37v7FGcJ
        ef9qr8/rQPh+RwzFpgdkHMUiEg==
X-Google-Smtp-Source: 
 APXvYqzY/NrXDMMzWYfoIwzHh3sPo0P5ORJ4j0cOgeIwUdZDRDI4SZtMfr6mDW9pMufCtCpKa+OJ4Q==
X-Received: by 2002:a1c:cfc3:: with SMTP id
 f186mr8407951wmg.134.1565335460541;
        Fri, 09 Aug 2019 00:24:20 -0700 (PDT)
Received: from hackbox2.linaroharston ([81.128.185.34])
        by smtp.gmail.com with ESMTPSA id
 p13sm26232705wrw.90.2019.08.09.00.24.19
        (version=TLS1_3 cipher=AEAD-AES256-GCM-SHA384 bits=256/256);
        Fri, 09 Aug 2019 00:24:19 -0700 (PDT)
From: Naresh Kamboju <naresh.kamboju@linaro.org>
To: naresh.kamboju@linaro.org, pbonzini@redhat.com, shuah@kernel.org
Cc: linux-kernel@vger.kernel.org, drjones@redhat.com,
        sean.j.christopherson@intel.com, linux-kselftest@vger.kernel.org,
        kvm@vger.kernel.org
Subject: [PATCH v3 1/2] selftests: kvm: Adding config fragments
Date: Fri,  9 Aug 2019 08:24:14 +0100
Message-Id: <20190809072415.29305-1-naresh.kamboju@linaro.org>
X-Mailer: git-send-email 2.17.1
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

selftests kvm all test cases need pre-required kernel config for the
tests to get pass.

CONFIG_KVM=y

The KVM tests are skipped without these configs:

        dev_fd = open(KVM_DEV_PATH, O_RDONLY);
        if (dev_fd < 0)
                exit(KSFT_SKIP);

Signed-off-by: Naresh Kamboju <naresh.kamboju@linaro.org>
Acked-by: Shuah Khan <skhan@linuxfoundation.org>
---
 tools/testing/selftests/kvm/config | 1 +
 1 file changed, 1 insertion(+)
 create mode 100644 tools/testing/selftests/kvm/config

diff --git a/tools/testing/selftests/kvm/config b/tools/testing/selftests/kvm/config
new file mode 100644
index 000000000000..14f90d8d6801
--- /dev/null
+++ b/tools/testing/selftests/kvm/config
@@ -0,0 +1 @@
+CONFIG_KVM=y

From patchwork Fri Aug  9 07:24:15 2019
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Naresh Kamboju <naresh.kamboju@linaro.org>
X-Patchwork-Id: 11085523
Return-Path: <kvm-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 1B5796C5
	for <patchwork-kvm@patchwork.kernel.org>;
 Fri,  9 Aug 2019 07:24:39 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id F2B1428C28
	for <patchwork-kvm@patchwork.kernel.org>;
 Fri,  9 Aug 2019 07:24:38 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id E6C4728C2D; Fri,  9 Aug 2019 07:24:38 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-8.0 required=2.0 tests=BAYES_00,DKIM_SIGNED,
	DKIM_VALID,DKIM_VALID_AU,MAILING_LIST_MULTI,RCVD_IN_DNSWL_HI
	autolearn=unavailable version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id C5B0E28C28
	for <patchwork-kvm@patchwork.kernel.org>;
 Fri,  9 Aug 2019 07:24:37 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S2405737AbfHIHYe (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Fri, 9 Aug 2019 03:24:34 -0400
Received: from mail-wr1-f66.google.com ([209.85.221.66]:45876 "EHLO
        mail-wr1-f66.google.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S2405701AbfHIHYd (ORCPT <rfc822;kvm@vger.kernel.org>);
        Fri, 9 Aug 2019 03:24:33 -0400
Received: by mail-wr1-f66.google.com with SMTP id q12so7006891wrj.12
        for <kvm@vger.kernel.org>; Fri, 09 Aug 2019 00:24:32 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=linaro.org; s=google;
        h=from:to:cc:subject:date:message-id:in-reply-to:references;
        bh=WA0xD70S4mvb/PQJqRGInCwCmljrweMt295QtIocBXo=;
        b=ECmtJohNEbHkaOvigVwh0bY5ueXewlnzoOQL3GT3J/vW0jaj8QP331qzTa6/eZe6ap
         FtgCKyWaI2sflGgqWmktF07KsIvaDVTZBEzGpnEFoqdjFHQJDHfA6TYzamqpL5yRiPdz
         uzOe419XsGH1DRGPvc6b1Wp8Z7lObTT7ntP5WtqGMZegXUN+mD85FpgToBqnZGaLapZj
         DjzTHVUT4HPBs4xXsvS1e4ie34ZZ7+rF/Et9jLybklI17im+xbwHvu4Hh3XmCSCbRa/f
         bFvlZ1E4nmQpCKjIJTF0SoLjYwmUZGLZp2NP/4mOtA88cRweUiVPLKMKKkVDkdHIRvC4
         1cSQ==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:from:to:cc:subject:date:message-id:in-reply-to
         :references;
        bh=WA0xD70S4mvb/PQJqRGInCwCmljrweMt295QtIocBXo=;
        b=I5VrjCWc4ntq4BG/HM18f2+1wfaTa5o58clIJNYV34GBDN9M1MzziK9xXtCRxaiHWq
         D7xRT8SK3foMCrXqb4TuYjqsNgDFQLuWIMtOIQxXg9PhnjE8FbEeFftdpATPMI2xLy4c
         2ISfiN1OcrUWPY2cC9rVNuKMq8HpVyazTrDNkBAqD3N098qocPf7LZw+dlTglRP6phF8
         RP0jT6QmacHYvWQUgPF2W5vtPZBwp5L34srlg4CzWH3HaMD94aqAj6zrGwfIUoJfem7t
         XRg++0/DosoBGVdNyFMd/hVzBhjmvtSauT1yuEzFI75wyk8ZzGJ85Js+P88MXULNHVDQ
         IDlg==
X-Gm-Message-State: APjAAAVqmUsn2SGuTTS2EWTf6RW2hGB6YnsPZY834SF2tYw6uUY7S9TP
        xj/V8GtJo6re7xXjgcGZVqUSmw==
X-Google-Smtp-Source: 
 APXvYqzeCO+GY7pXqejgTG/gXxeki0mTFaAubI9/NXbDdkGSDD2/esm1gSsnUs44X+UO9i0yvJ2edQ==
X-Received: by 2002:a5d:4484:: with SMTP id j4mr22016063wrq.143.1565335471654;
        Fri, 09 Aug 2019 00:24:31 -0700 (PDT)
Received: from hackbox2.linaroharston ([81.128.185.34])
        by smtp.gmail.com with ESMTPSA id
 p13sm26232705wrw.90.2019.08.09.00.24.30
        (version=TLS1_3 cipher=AEAD-AES256-GCM-SHA384 bits=256/256);
        Fri, 09 Aug 2019 00:24:31 -0700 (PDT)
From: Naresh Kamboju <naresh.kamboju@linaro.org>
To: naresh.kamboju@linaro.org, pbonzini@redhat.com, shuah@kernel.org
Cc: linux-kernel@vger.kernel.org, drjones@redhat.com,
        sean.j.christopherson@intel.com, linux-kselftest@vger.kernel.org,
        kvm@vger.kernel.org
Subject: [PATCH v3 2/2] selftests: kvm: x86_64: Adding config fragments
Date: Fri,  9 Aug 2019 08:24:15 +0100
Message-Id: <20190809072415.29305-2-naresh.kamboju@linaro.org>
X-Mailer: git-send-email 2.17.1
In-Reply-To: <20190809072415.29305-1-naresh.kamboju@linaro.org>
References: <20190809072415.29305-1-naresh.kamboju@linaro.org>
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

selftests kvm x86_64 test cases need pre-required kernel configs for the
tests to get pass when you are using Intel or AMD CPU.

CONFIG_KVM_INTEL=y
CONFIG_KVM_AMD=y

Signed-off-by: Naresh Kamboju <naresh.kamboju@linaro.org>
---
 tools/testing/selftests/kvm/x86_64/config | 2 ++
 1 file changed, 2 insertions(+)
 create mode 100644 tools/testing/selftests/kvm/x86_64/config

diff --git a/tools/testing/selftests/kvm/x86_64/config b/tools/testing/selftests/kvm/x86_64/config
new file mode 100644
index 000000000000..4df8c7f54885
--- /dev/null
+++ b/tools/testing/selftests/kvm/x86_64/config
@@ -0,0 +1,2 @@
+CONFIG_KVM_INTEL=y
+CONFIG_KVM_AMD=y
